File: corporate-reporting/src/types/WorkStream.ts
import { IReportingEntity, ReportingEntity } from './ReportingEntity';
import { IProject } from './Project';
import { IUser } from './User';
import { IMilestone } from './Milestone';
import { IWorkStreamUpdate } from './WorkStreamUpdate';

export interface IWorkStream extends IReportingEntity {
    WorkStreamCode: string;
    ProjectID: number;
    LeadUserID: number;
    RagOptionID: number;
    Project?: IProject;
    LeadUser?: IUser;
    Milestones?: IMilestone[];
    WorkStreamUpdates?: IWorkStreamUpdate[];
}

export class WorkStream extends ReportingEntity implements IWorkStream {
    public WorkStreamCode = '';
    public ProjectID = null;
    public LeadUserID = null;
    public RagOptionID = null;
}

File: ORB.Core/Models/WorkStream.cs
using System;
using System.Collections.Generic;

namespace ORB.Core.Models
{
    public class WorkStream : ReportingSubEntity
    {
        public WorkStream()
        {
            Milestones = new HashSet<Milestone>();
            WorkStreamUpdates = new HashSet<WorkStreamUpdate>();
        }

        public string WorkStreamCode { get; set; }
        public int ProjectID { get; set; }
        public int? RagOptionID { get; set; }

        public virtual Project Project { get; set; }
        public virtual RagOption RagOption { get; set; }
        public virtual ICollection<Milestone> Milestones { get; set; }
        public virtual ICollection<WorkStreamUpdate> WorkStreamUpdates { get; set; }
    }
}
